### ISSUE TITLE
RabbitMQ Dashboard is not installed with service

### ISSUE THREAD

**Issue by mikepc** [vmware-archive/pcfdev#277]
See my Stack Overflow post here:

https://stackoverflow.com/questions/48106204/pcf-dev-rabbitmq-management-dashboard-not-installed

Any help on what step I'm missing would be great!

**Comment by sclevine** [comment]
Hi @mikepc, I'm sorry you're running into trouble with the rabbitmq dashboard URL. The URL should just work. I can test it myself when I get back from vacation, but these questions may help you out in the meantime:
- Have you tried destroying the VM and starting it again?
- Have you tried accessing the dashboard URL after binding the service to an app? 
- Could you see if it works on an older version of PCF Dev, such as 0.21.0? We specifically added support for the rabbitmq dashboard URL in 0.21.0.

We're currently re-writing PCF Dev to use a real BOSH director and run on the native OS hypervisors. We have no plans to release further versions of the current implementation. If this is a regression, it is unlikely to be fixed until we are able to release the re-write. Sorry for the trouble.

Regarding https://github.com/pivotal-cf/docs-pcf-dev/issues/38: I have no idea who owns pcfdev.slack.com. I suspect it was registered by past team members and is now unused. If you'd like to reach us on Slack, we have a #pcfdev channel in the Cloud Foundry slack.

**Comment by mikepc** [comment]
I have definitely tried the first 2! I've tried explicitly specifying rabbitmq during the install process as well.
I'm going to go try an earlier version of pcf dev right now!

**Comment by mikepc** [comment]
Confirmed. I rolled back to **pcfdev-v0.26.0+PCF1.10.0-osx and the problem is resolved**, I'm able to bring up the RabbitMQ management console.  Later versions *may* work, but I went down to the last minor version (down to 1.10 from 1.11)

I can also confirm that _pcfdev-v0.28.0+PCF1.11.0-osx_ **does not work** for the RabbitMQ management console.

Also I haven't tried any other OS's, the only installers I've worked with were 28( didn't work) and 26 (Worked)

**Comment by jrbrayjr2006** [comment]
I am seeing the same issue on the latest version of PCFDev.  I am using the cloud version for development to get around this for the time being.  This does slow things down a bit.

**Comment by mikepc** [comment]
I'm curious what the manual steps would be to install the route?

**Comment by dcarwin-pivotal** [comment]
Product leadership pinged for an update on this issue. (2018-03-01)

**Comment by sclevine** [comment]
@mikepc @dcarwin-pivotal Development on PCF Dev has moved [here](https://github.com/pivotal-cf/cfdev), where we're working on an a new version that is being reviewed for incubation in the Cloud Foundry Foundation. All new releases of PCF Dev will be via an extension to that version. While we do not yet have a timeline for the PCF extension or for rabbitmq integration, we do have an alpha release available that will allow you to deploy the cf-rabbitmq BOSH release manually.

@ssisil is currently testing whether that integration works, and will report back here shortly with instructions. Note that the alpha release requires macOS Sierra or greater, and currently fails to start without Docker for Mac installed, due to a bug. (On the plus side, it doesn't require Virtualbox and comes with a BOSH director!)

It's worth noting that PCF Dev is a free beta product, not a supported Pivotal product. If you're a PCF customer using PCF Dev in a way that is critical to your business, please let us know!

**Comment by ssisil** [comment]
@mikepc @dcarwin-pivotal 

I was able to get the [RabbitMQ ](https://github.com/pivotal-cf/cf-rabbitmq-release)bosh release running with our alpha version of [CF Dev](https://github.com/pivotal-cf/cfdev).  

You need to make a few tweaks to the manifest and lite vars file in the release to get things up and running :

__lite-vars-file.yml__
--> Change stemcell-version to `3.468.17` (the one bundled with CF Dev)
--> Change the bosh-domain to `v3.pcfdev.io`

__cf-rabbitmq-template.yml__
--> change vm_type to `small-highmem` for` rmq` instance_group
--> change vm_type to `small-highmem` for `haproxy` instance_group

I would recommend uploading the release from bosh.io versus trying to create the release from source
```
# Upload latest version, currently 238.0.0
$ bosh upload-release https://bosh.io/d/github.com/pivotal-cf/cf-rabbitmq-release --sha1 7a88d3cda9f92b34f28f056cf2c64f70f8546b7c
```

Then generate your manifest.yml using the `scripts/generate-manifest` script included in the release source code.  Once you have your manifest.yml created and the bosh release uploaded - simply deploy the release using the following command - `bosh -d cf-rabbitmq deploy <path to manifest.yml>`.

Once it is deployed you should be able to access the rabbitMQ manager at `http://rabbitmq.v3.pcfdev.io`.

Let us know how it goes.

**Comment by mikepc** [comment]
Hi Stephen, we are a baby company that hasn’t even formed yet.  There are
three engineers on the project (5 in the company). and we really don’t need
a cloud deployment, but our plan is to deploy to PWS in May or June with a
non-prod and prod environment.  PCF Dev is working awesome for us during
incubation.

The rabbitmq issue isn’t a blocker but it is.. annoying.  One of my
partners recommended it and we haven’t regretted it yet.

Please don’t abandon the project! Pretty please!

It is the cleanest, simplest, and fastest way to ramp up a simple
microservice architecture during incubation.

Minikube takes a lot more hassle to get ramped up with and requires all of
the engineers to understand k8s.  As far as I am concerned it’s the only
other realistic option.

**Comment by sclevine** [comment]
I'm glad you find PCF Dev so compelling! I don't think your use case is unreasonable, as long as you aren't using PCF Dev to host production apps that others outside of your group have access to.

We have no plans to abandon PCF Dev, just plans to update the underlying implementation to use [CF Dev](https://github.com/pivotal-cf/cfdev) with PCF patches and extras.

**Comment by dgcarlso3** [comment]
I am having the same issue with rabbitmq console. I tried to install pcfdev-v0.26.0+PCF1.10.0-osx as the above user, and cannot start pcf dev as per this issue: https://github.com/pivotal-cf/pcfdev/issues/273. How can I get this version started and get up and running with pcfdev?

**Comment by sclevine** [comment]
There are two options:
1. Deploy rabbitmq to CF Dev, using the instructions that @ssisil provided.
2. Set your system clock back several months before starting PCF Dev v0.26.0. (**I strongly discourage this option. Bad things may happen.**)

**Comment by airduster** [comment]
Having this same issue, I was thinking we could maybe find a "hack" solution for the management console using the utility 'rabbitmqctl' that ships with rabbitmq in the bin dir from a separate local installation. This utility can be used to start/stop management console.

My question Is there a way to talk to the rabbitmq instance inside VM? I came close to achieving this by accessing the rabbitmq instance generated by pcfdev. I was able to connect via a local installation's 'rabbitmqctl' utility but after successfully establishing a connection it was rejected because I didn't have the .erlang.cookie file needed to authenticate connecting from a different host (this file located is inside the user directory in the pcfdev generated VirtualBox .vmdk file) see screenshot. 

I am thinking if we can somehow obtain this .erlang.cookie file/value we can use a local 'rabbitmqctl' utility to spin up the management console? Is there any easy way to access the PCFDev generated VM file system? Right now i'm stuck at this point I cannot obtain access to the .erlang.cookie file in the VM to test to see if this will work. Also, I can ping the host and resolve the hostname which leads me to suspect the rabbitmq management console just needs to be started manually.  

![image](https://user-images.githubusercontent.com/2158763/41517193-b55ed540-727e-11e8-8818-f8c578b32ed2.png)
